Build error on MSVC 2017

I get the following build error when I try to enable tests and examples on MSVC 2017 x64:
```
d:\local\builds\highfive\include\highfive\bits\H5Converter_misc.hpp(90): error C2668: 'HighFive::details::vectors_to_single_buffer': ambiguous call to overloaded function
d:\local\builds\highfive\include\highfive\bits\H5Converter_misc.hpp(82): note: could be 'void HighFive::details::vectors_to_single_buffer<double>(const std::vector<double,std::allocator<_Ty>> &,const std::vector<::size_t,std::allocator<unsigned __int64>> &,::size_t,std::vector<_Ty,std::allocator<_Ty>> &)'
        with
        [
            _Ty=double
        ]
d:\local\builds\highfive\include\highfive\bits\H5Converter_misc.hpp(70): note: or       'void HighFive::details::vectors_to_single_buffer<double>(const std::vector<double,std::allocator<_Ty>> &,const std::vector<::size_t,std::allocator<unsigned __int64>> &,const ::size_t,std::vector<_Ty,std::allocator<_Ty>> &)'
        with
        [
            _Ty=double
        ]
d:\local\builds\highfive\include\highfive\bits\H5Converter_misc.hpp(90): note: while trying to match the argument list '(const std::vector<double,std::allocator<_Ty>>, const std::vector<::size_t,std::allocator<_Ty>>, ::size_t, std::vector<double,std::allocator<_Ty>>)'
        with
        [
            _Ty=double
        ]
        and
        [
            _Ty=::size_t
        ]
        and
        [
            _Ty=double
        ]
d:\local\builds\highfive\include\highfive\bits\H5Converter_misc.hpp(382): note: see reference to function template instantiation 'void HighFive::details::vectors_to_single_buffer<T>(const std::vector<std::vector<double,std::allocator<_Ty>>,std::allocator<std::vector<_Ty,std::allocator<_Ty>>>> &,const std::vector<::size_t,std::allocator<unsigned __int64>> &,::size_t,std::vector<_Ty,std::allocator<_Ty>> &)' being compiled
        with
        [
            T=std::vector<double,std::allocator<double>>,
            _Ty=double
        ]
```
I'm not sufficiently familiar with HighFive to debug this myself, can anyone help?

--- a/include/highfive/bits/H5Converter_misc.hpp
+++ b/include/highfive/bits/H5Converter_misc.hpp
@@ -73,17 +73,17 @@ inline void vectors_to_single_buffer(const std::vector<T>& vec_single_dim,
     buffer.insert(buffer.end(), vec_single_dim.begin(), vec_single_dim.end());
 }
 
-template <typename T>
+template <typename T, typename U = typename type_of_array<T>::type>
 inline void
 vectors_to_single_buffer(const std::vector<T>& vec_multi_dim,
-                         const std::vector<size_t>& dims, size_t current_dim,
-                         std::vector<typename type_of_array<T>::type>& buffer) {
+                         const std::vector<size_t>& dims,
+                         size_t current_dim,
+                         std::vector<U>& buffer) {
 
     check_dimensions_vector(vec_multi_dim.size(), dims[current_dim],
                             current_dim);
-    for (typename std::vector<T>::const_iterator it = vec_multi_dim.begin();
-         it < vec_multi_dim.end(); ++it) {
-        vectors_to_single_buffer(*it, dims, current_dim + 1, buffer);
+    for (const auto& it : vec_multi_dim) {
+        vectors_to_single_buffer(it, dims, current_dim + 1, buffer);
     }
 }
 

